### PR TITLE
Various additions and corrections for window

### PR DESCRIPTION
**Pull request by Zirro**
These are a collection of changes I made while exploring the accuracy of our `window` implementation.

* Implement the `BarProp` interface and the `window` attributes that use it.

* Implement the `Screen` interface used for `window.screen`, which was previously just an object with two properties instead of the six attributes it actually contains. Tested in `cssom-view-window-screen-interface.html` and `Screen-pixelDepth-Screen-colorDepth001.html`.

* Implement `window.devicePixelRatio` with the default value `1`.

* Implement `window.status` which exists for historical reasons.

* Expose the `External` interface.

### CODE REVIEW

File: lib/jsdom/living/window/Screen-impl.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+// https://drafts.csswg.org/cssom-view-1/#the-screen-interface
+exports.implementation = class ScreenImpl {
+  get availWidth() {
+    return 0;
```
Comment (TimothyGu) [inline lib/jsdom/living/window/Screen-impl.js]: These will never get changed, so I'd rather just do something like https://github.com/TimothyGu/jsdom/blob/window-webidl2js/lib/jsdom/living/window/Screen-impl.js

```js
class ScreenImpl {}
ScreenImpl.prototype.availWidth = 0;
ScreenImpl.prototype.availHeight = 0;
ScreenImpl.prototype.width = 0;
ScreenImpl.prototype.height = 0;
ScreenImpl.prototype.colorDepth = 0;
ScreenImpl.prototype.pixelDepth = 0;
```

File: lib/jsdom/living/window/Screen-impl.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+// https://drafts.csswg.org/cssom-view-1/#the-screen-interface
```
Comment (TimothyGu) [inline lib/jsdom/living/window/Screen-impl.js]: I think we usually have such URL comments in the IDL file. This is totally fine by me, but just wanted to know if that policy changed..

Comment (Zirro) [reply lib/jsdom/living/window/Screen-impl.js]: I believe they're mainly used for partials within IDL files. That said, I would not mind making it standard practice to have them in both IDL and the implementation. They're often handy for looking up implementation details in the current context.

File: lib/jsdom/living/window/Screen-impl.js (hunk 1 above)
Comment (Zirro) [reply lib/jsdom/living/window/Screen-impl.js]: Oh, I wasn't aware of how far you've gotten on the new implementation of window. The refactored version looks great!

Regarding this file in particular, I wouldn't mind making this change. I usually prioritise being consistent across files, but the advantages of making an exception here are pretty clear.

Comment (Sebmaster) [reply lib/jsdom/living/window/Screen-impl.js]: I'd also prefer the prototype version for static values (wherever possible).

File: lib/jsdom/living/window/Screen-impl.js (hunk 2 above)
Comment (Sebmaster) [reply lib/jsdom/living/window/Screen-impl.js]: I think we usually do IDL source link in the IDL file, implementation detail links in impl.

File: lib/jsdom/browser/Window.js [hunk 3]
```diff
@@ -538,6 +538,7 @@ function Window(options) {
 
   define(this, {
     name: "nodejs",
+    devicePixelRatio: 1,
```
Comment (domenic) [inline lib/jsdom/browser/Window.js]: Note that I think these should all be getters (own-getters, but still getters). Happy to continue with them in their current state for now though.

File: lib/jsdom/living/window/Screen-impl.js (hunk 1 above)
Comment (domenic) [reply lib/jsdom/living/window/Screen-impl.js]: I kind of like the consistency, given that these aren't going to be commonly accessed. But either way.

**Comment by domenic** [comment]
It looks like the addition of `status` is causing XHR test failures on Node v6 due to some vm stuff. We should add needs-node8 to those tests.

**Comment by Zirro** [comment]
Updated to address feedback. I suppose we'll have to decide on a final getter vs prototype-approach later on.

> It looks like the addition of status is causing XHR test failures on Node v6 due to some vm stuff. We should add needs-node8 to those tests.

Since `window.status` isn't that important, I've decided place it behind a comment with a note until we stop supporting Node v6. After all, if it causes issues in a test there is a risk that it could break real uses too.

File: lib/jsdom/living/index.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ exports.WheelEvent = require("./generated/WheelEvent").interface;
 exports.EventTarget = require("./generated/EventTarget").interface;
 
 exports.BarProp = require("./generated/BarProp").interface;
+exports.External = require("./generated/External").interface;
```
Comment (domenic) [inline lib/jsdom/living/index.js]: So actually `External` is `[NoInterfaceObject]` so we should not do this. I guess we should try to make use of the `exposed` thingy some day.

Comment (TimothyGu) [reply lib/jsdom/living/index.js]: > I guess we should try to make use of the `exposed` thingy some day.

Already have a WIP :)

**Pull request merged**